### PR TITLE
Dynamically create schedule which will be added to the user set of calendars on Google Calendar

### PR DESCRIPTION
**Pull request by JabariD**
The biggest things to note in this PR is:
1) SchedulerGenerationServlet.java. -- The MAIN Servlet in which we generate the schedule
2) UserPreferences.java is where we will UPDATE the user settings.
3) Time.java is where we interact with the TIME which is used throughout SchedulerGenerationServlet.java
4) and a bunch of different resource gatherers, simply to allow easier interaction with the Google Calendar API

REMEMBER: For now, each time we go to '/request-permission' we will generate this schedule and IT WILL APPEAR ON YOUR CALENDARS (don't forget to delete after done testing). This is just for the MVP.

@chrisc94 @maricarol3355 @cjhbh3 @ptai7

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/schedule/ScheduleGenerationServlet.java [hunk 1]
```diff
@@ -78,7 +79,67 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
...
+    String timezone = (String) timeZoneSetting.get("value");
+    TIME.setTimeZoneId(timezone);
+
+    // Let's try to create events! We depend on RECURRING EVENTS. So, upto the next week, we try to create 
+    Integer successfulEventsCreated = 0;
+
+    // Set user calendars to get FreeBusy info here
+    List<String> ids = getAllCalendarIds(httpClient, accessToken);
+
+    Integer stepValue = USER.EVENT_LOOK_SPAN / 2;
```
Comment (chrisc94) [inline src/main/java/com/google/sps/servlets/schedule/ScheduleGenerationServlet.java:98]: Why not just 'int'?

Java is able to convert between the two easily and the raw type int is usually preferred

File: src/main/java/com/google/sps/servlets/schedule/util/Time.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import java.util.List;
+import java.util.Map;
+
+import org.joda.time.DateTimeZone;
+import org.joda.time.DateTime;
+
+/** Should be able to READ string. */
+public class Time {
+    // For recurrences!
+    public final String SUNDAY = "SU";
```
Comment (chrisc94) [inline src/main/java/com/google/sps/servlets/schedule/util/Time.java:22]: Maybe we can declare a Map<Integer, String> where the key = 1 to 7 mapped to "SU", "MO", "TU" so on.

You can initialize the map when you create it: https://stackoverflow.com/questions/6802483/how-to-directly-initialize-a-hashmap-in-a-literal-way

Then you can replace the case statement with a map lookup.

File: src/main/java/com/google/sps/servlets/schedule/ScheduleGenerationServlet.java (hunk 1 above)
Comment (JabariD) [reply src/main/java/com/google/sps/servlets/schedule/ScheduleGenerationServlet.java:98]: Right!

File: src/main/java/com/google/sps/servlets/schedule/util/Time.java (hunk 2 above)
Comment (JabariD) [reply src/main/java/com/google/sps/servlets/schedule/util/Time.java:22]: Makes sense!

**Comment by maricarol3355** [comment]
LGTM Payton!

File: src/main/java/com/google/sps/servlets/schedule/ScheduleGenerationServlet.java [hunk 3]
```diff
@@ -78,7 +79,67 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
...
+      } 
+
+      // Increase step value
+      day += stepValue;
+    }
+
+    //////////////////////// 4) Call Fixer | ALTER BY USER SETTING: just delete event / find next available time / force move to next day /
+    // the fixer should be given a list of days and then perform user action
+
+		httpClient.getConnectionManager().shutdown();
```
Comment (ptai7) [inline src/main/java/com/google/sps/servlets/schedule/ScheduleGenerationServlet.java:142]: nit: De-indent.

File: src/main/java/com/google/sps/servlets/schedule/ScheduleGenerationServlet.java [hunk 4]
```diff
@@ -95,4 +156,218 @@ JSONObject createNewCalendar(DefaultHttpClient httpClient, String accessToken, S
...
+      } catch (Exception e) {
+        System.out.println("An exception occurred!" + e);
+        continue;
+      }
+    }
+
+    // What are we trying to do?
+    // We trying to determine if a time period is valid. So we loop through each time period and it's duration and check it against each busy period.
+    // After it's been checked against every busy period and we haven't continued, then it's valid!
+    // NOTE: We add
```
Comment (ptai7) [inline src/main/java/com/google/sps/servlets/schedule/ScheduleGenerationServlet.java:263]: nit: finish comment?

File: src/main/java/com/google/sps/servlets/schedule/UserPreferences.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+    // How long should we repeat this event
+    public static int EVENT_RECURRENCE_LENGTH = 4; // in weeks
+
+    // Possible study event sessions
+    public Map<Integer, Integer> STUDY_SESSION_START_TIME = new LinkedHashMap<Integer, Integer>();
+    public Map<Integer, Integer> STUDY_SESSION_LENGTH = new LinkedHashMap<Integer, Integer>();
+
+    // Set description if no resources could be found
+    public static String DESCRIPTION = "Read for motivation: https://medium.com/dev-genius/why-you-should-stop-watching-tutorials-and-reading-docs-escaping-tutorial-purgatory-2c6b438b418 \n\n Best recommended free site: https://www.codecademy.com/\n\nBest recommended paid site: https://www.udemy.com/";
```
Comment (ptai7) [inline src/main/java/com/google/sps/servlets/schedule/UserPreferences.java:45]: Great addition! :)

**Pull request merged**